### PR TITLE
feat: metric is now a generic algo

### PR DESCRIPTION
**Pull request by Esadruhn**
The `Metrics` class:
- now inherits from the `GenericAlgo`
- is renamed `MetricAlgo`
- the CLI args are the same as the other algos (inputs, outputs)

Companion PRs on:
- substra: https://github.com/Substra/substra/pull/290
- substra-backend: https://github.com/Substra/substra-backend/pull/472
- substra-tests: https://github.com/Substra/substra-tests/pull/214
- substrafl: https://github.com/Substra/substrafl/pull/24
- substra-documentation: https://github.com/Substra/substra-documentation/pull/183

### CODE REVIEW

**Comment by Esadruhn** [comment]
/e2e --refs substra-tools=metrics_to_generic,substra=metrics_to_generic,substra-tests=metrics_to_generic,substrafl=metrics_to_generic,connect-ci=fix/substra_tools_checkout --tests sdk,substrafl --benchmarks mnist,camelyon

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by Esadruhn** [comment]
e2e tests error: "AttributeError: module 'substratools' has no attribute 'MetricAlgo'"

**Comment by Esadruhn** [comment]
/e2e --refs substra-tools=metrics_to_generic,substra=metrics_to_generic,substra-tests=metrics_to_generic,substrafl=metrics_to_generic,connect-ci=fix/substra_tools_checkout --tests sdk,substrafl --mode standalone

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by Esadruhn** [comment]
built 3.9 image instead of 3.8

**Comment by Esadruhn** [comment]
/e2e --refs substra-tools=metrics_to_generic,substra=metrics_to_generic,substra-tests=metrics_to_generic,substrafl=metrics_to_generic,connect-ci=fix/substra_tools_checkout --tests sdk,substrafl --mode standalone

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by Esadruhn** [comment]
```json
{"directory": "{\"name\": \"task\", \"type\": \"directory\", \"children\": [{\"name\": \"chainkeys\", \"type\": \"directory\", \"children\": []}, {\"name\": \"in_models\", \"type\": \"directory\", \"children\": [{\"name\": \"0c2dd39a-26│
│{"namespace": "org-1", "name": "substra-fd91929b-compute-algo-82f", "event": "Creating pod", "compute_plan_key": "fd91929b-4052-4b44-9c98-b8cfc2993009", "compute_task_key": "fa7ce11e-5ac6-4afe-ab8f-18aab6fa9aa5", "attempt": 1, "task_id"│
│{"command": ["python3", "-u", "metrics.py", "--method-name", "score", "--input-predictions-path", "/substra_internal/in_models/0c2dd39a-26ab-44cd-9cc4-e02b8d5cfc36", "--opener-path", "/substra_internal/openers/a45bd477-17fa-407f-83f9-7c│
│{"event": "usage: metrics.py [-h] [--method-name METHOD_NAME] [-r RANK] [-d]", "compute_plan_key": "fd91929b-4052-4b44-9c98-b8cfc2993009", "compute_task_key": "fa7ce11e-5ac6-4afe-ab8f-18aab6fa9aa5", "attempt": 1, "task_id": "fa7ce11e-5a│
│{"event": "                  [--n-fake-samples N_FAKE_SAMPLES] [--log-path LOG_PATH]", "compute_plan_key": "fd91929b-4052-4b44-9c98-b8cfc2993009", "compute_task_key": "fa7ce11e-5ac6-4afe-ab8f-18aab6fa9aa5", "attempt": 1, "task_id": "fa7│
│{"event": "                  [--log-level {debug,info,warning,error,critical}]", "compute_plan_key": "fd91929b-4052-4b44-9c98-b8cfc2993009", "compute_task_key": "fa7ce11e-5ac6-4afe-ab8f-18aab6fa9aa5", "attempt": 1, "task_id": "fa7ce11e-│
│{"event": "                  [--inputs INPUTS] [--outputs OUTPUTS]", "compute_plan_key": "fd91929b-4052-4b44-9c98-b8cfc2993009", "compute_task_key": "fa7ce11e-5ac6-4afe-ab8f-18aab6fa9aa5", "attempt": 1, "task_id": "fa7ce11e-5ac6-4afe-ab│
│{"event": "metrics.py: error: unrecognized arguments: --input-predictions-path /substra_internal/in_models/0c2dd39a-26ab-44cd-9cc4-e02b8d5cfc36 --opener-path /substra_internal/openers/a45bd477-17fa-407f-83f9-7c3038b76ae5/__init__.py --d│
│{"function": "_exec", "duration": "283.55ms", "event": "(profiler) function returned an error", "compute_plan_key": "fd91929b-4052-4b44-9c98-b8cfc2993009", "compute_task_key": "fa7ce11e-5ac6-4afe-ab8f-18aab6fa9aa5", "attempt": 1, "task_│
│{"function": "execute_compute_task", "duration": "14233.05ms", "event": "(profiler) function returned an error", "compute_plan_key": "fd91929b-4052-4b44-9c98-b8cfc2993009", "compute_task_key": "fa7ce11e-5ac6-4afe-ab8f-18aab6fa9aa5", "at│
│
```

**Comment by Esadruhn** [comment]
/e2e --refs substra-tools=metrics_to_generic,substra-backend=metrics_to_generic,substra=metrics_to_generic,substra-tests=metrics_to_generic,substrafl=metrics_to_generic,connect-ci=fix/substra_tools_checkout --tests sdk,substrafl --mode standalone

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

File: substratools/algo.py [hunk 1]
```diff
@@ -229,6 +232,28 @@ def predict(
...
+    with open(path, "w") as f:
+        json.dump({"all": performance}, f)
+
+
+def load_performance(path: os.PathLike) -> Any:
+    with open(path, "r") as f:
+        performance = json.load(f)["all"]
+    return performance
+
+
```
[the remark this answers is not in the document]
Comment (Esadruhn) [reply substratools/algo.py:265]: it is done in the backend, and in substratools too, see the metrics tests we try to cast it to a float

**Pull request merged**